subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/internal/toolchains/DefaultToolChainSelector.java

```diff
@@ -109,6 +109,8 @@ static SwiftVersion toSwiftVersion(VersionNumber swiftCompilerVersion) {
             return SwiftVersion.SWIFT3;
```

Comment:
Just to remove one bit of maintenance...

```
    static SwiftVersion toSwiftVersion(VersionNumber swiftCompilerVersion) {
        for (SwiftVersion version : SwiftVersion.values()) {
            if (version.getVersion() == swiftCompilerVersion.getMajor()) {
                return version;
            }
        }
        throw new IllegalArgumentException(String.format("Swift language version is unknown for the specified Swift compiler version (%s)", swiftCompilerVersion.toString()));
    }
```

Reply:
Done.